### ISSUE TITLE
Incorrect SCF energys for some STRU using the version 2.0 TZDP orbitals

### ISSUE THREAD

**Issue by wszhang** [abacusmodeling/abacus-develop#212]
### Describe the bug

By using the new version2.0 TZDP 8au orbital, several lower SCF energys can be obtained compared with the results of PW calculations for STRUs with shorter bond-length:
![21_Sc_E100_TZDP_add_1z1p_BE VE](https://user-images.githubusercontent.com/10769183/195996610-f393f2ae-7df6-4945-8f0f-bd47d0d2ff21.png)

The results are reproducible with ABACUS v2.0.0, v2.2.2 and v2.3.0.

When using DZP orbitals, the results of SCF are correct:
![21_Sc_E100_DZP_add_1z1p_BE VE](https://user-images.githubusercontent.com/10769183/195996635-abd85f4f-5b3b-450d-8a5f-6b6d245915d2.png)

### Expected behavior

For version 2.0 orbitals, each TZDP orbital-file includes the orbitals of related DZP orbital-file, one should obtain good VE results by using the TZDP orbitals if good VE results had been obtained by using DZP orbitals.

### To Reproduce

The input files:
[input_files.zip](https://github.com/abacusmodeling/abacus-develop/files/9792564/input_files.zip)

### Environment

Hanhai20 @ USTC:
module load abacus/2.3.0/intel-2019.update5
or
module load abacus/2.2.2/intelmpi2018u4_elpa202105
or
module load abacus/2.0.0/intel-2018.update4

**Comment by dyzheng** [comment]
I will test this case,

**Comment by dyzheng** [comment]
This is the consequence of "ecutwfc 150"
```
 -------------------------------------------
 SELF-CONSISTENT : 
 -------------------------------------------
 ITER   ETOT(eV)       EDIFF(eV)      DRHO       TIME(s)    
 GE1    -2.499928e+03  0.000000e+00   5.361e-02  2.950e+01  
 GE2    -2.500227e+03  -2.994882e-01  1.584e-02  2.835e+01  
 GE3    -2.500249e+03  -2.207239e-02  2.312e-03  2.837e+01  
 GE4    -2.500249e+03  -1.773934e-05  2.172e-04  2.832e+01  
 GE5    -2.500249e+03  -2.268267e-06  2.987e-05  2.830e+01  
 GE6    -2.500249e+03  -1.347910e-08  1.448e-06  2.835e+01  
 GE7    -2.500249e+03  4.783444e-10   2.235e-07  2.835e+01  
 GE8    -2.500249e+03  4.802779e-10   4.733e-08  2.833e+01  
 GE9    -2.500249e+03  5.247481e-10   5.386e-09  2.834e+01
```
which is same as PW base, and different with "ecutwfc 100"
```
-------------------------------------------
 SELF-CONSISTENT : 
 -------------------------------------------
 ITER   ETOT(eV)       EDIFF(eV)      DRHO       TIME(s)    
 GE1    -2.512224e+03  0.000000e+00   1.578e-01  3.447e+01  
 GE2    -2.531765e+03  -1.954065e+01  1.225e-01  3.387e+01  
 GE3    -2.547875e+03  -1.611004e+01  6.330e-02  3.404e+01  
 GE4    -2.559166e+03  -1.129060e+01  2.845e-02  3.395e+01  
 GE5    -2.562730e+03  -3.564374e+00  1.210e-01  3.420e+01  
 GE6    -2.562944e+03  -2.141839e-01  1.155e-02  3.425e+01  
 GE7    -2.565775e+03  -2.830998e+00  5.314e-03  3.408e+01  
 GE8    -2.567880e+03  -2.104520e+00  2.630e-03  3.410e+01  
 GE9    -2.567825e+03  5.514962e-02   1.281e-03  3.402e+01  
 GE10   -2.567840e+03  -1.536272e-02  6.653e-04  3.418e+01  
 GE11   -2.567729e+03  1.106852e-01   9.191e-05  3.407e+01  
 GE12   -2.567751e+03  -2.137925e-02  8.578e-05  3.403e+01  
 GE13   -2.567752e+03  -1.244847e-03  1.315e-05  3.410e+01  
 GE14   -2.567751e+03  4.202577e-04   2.577e-06  3.420e+01  
 GE15   -2.567751e+03  6.378731e-04   5.085e-07  3.414e+01  
 GE16   -2.567751e+03  -1.291552e-04  1.436e-07  3.396e+01  
 GE17   -2.567751e+03  -2.142204e-06  1.838e-08  3.404e+01  
 GE18   -2.567751e+03  -1.386714e-06  6.975e-09  3.415e+01
```

**Comment by jinzx10** [comment]
I did a detailed analysis of the basis orbitals and the results do support @dyzheng 's observation that increasing the energy cutoff for 1-d integration to 150 Ry would solve the issue. 

The plot below shows the f orbitals of the two basis sets. The first row corresponds to DZP and the second row TZDP. The 3 figures in each row are the radial function, its k-space counterpart, and the integrand in the kinetic integral. The energy cutoffs (100 & 150 Ry) are converted to k cutoffs and marked in the figures.

The figure shows that, while 100 Ry looks OK for k-space function themselves, it is not adequate for doing kinetic integrals where the k^4 factor significantly magnifies the tail. As a result, calculations with 100 Ry cutoff significantly underestimate the kinetic energy and the result is dubious. 

The reason why DZP basis set does not fail with 100 Ry cutoff is probably because the problematic f orbital is basically not occupied, so its problematic kinetic energy does not participate in the final energy calculation. On the contrary, in TZDP calculation, some f orbitals are somewhat occupied, therefore their problematic kinetic energy takes part in the final energy.

Since 1D integration seems to be the only issue and 100 Ry is already a reasonably good energy cutoff for grid integrals, I would recommend keeping "ecutwfc 100", and setting "lcao_ecut 150" (or some even larger number to check convergence).

PS: the occupation is checked for gamma-only calculation where the final energy is even more erratic. For the multi-k calculation, ABACUS currently does not support dumping the density matrix to file and I did not find an easy way to do this. Also I do not fully understand why the f orbital in DZP calculation is basically unoccupied yet some of the f orbitals in TZDP are occupied. But I think increasing lcao_ecut to cover the tail of k^4 chi^2(k) is the solution.

![e39ee616-2815-42df-af3b-0450399c0d52](https://user-images.githubusercontent.com/26805483/234314067-42fcae3c-7f22-48cb-9dc9-5db9887d91ad.jpeg)
